OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -152,7 +152,8 @@ protected virtual bool CanAttack(Actor self, in Target target)
 			if (!HasAnyValidWeapons(target))
 				return false;
 
-			var mobile = self.TraitOrDefault<Mobile>();
+			// PERF: Mobile implements IFacing, so we can use 'as' to save a trait look-up here.
```

Comment:
Minor style nit: IMO it feels more natural to use `IPositionable` instead of `IFacing` here, but since it doesn't really matter for practical purposes I don't consider this to be a blocker.